src/libAtomVM/atom.c

```diff
@@ -55,20 +55,32 @@ int atom_are_equals(AtomString a, AtomString b)
 
 void atom_write_mfa(char *buf, size_t buf_size, AtomString module, AtomString function, int arity)
 {
+
+    unsigned int arity_num_digits = 0;
+    // Simple calculation because we don't expect arities greater than 255
+    if (arity < 10) {
+        arity_num_digits = 1;
+    } else if (arity < 100) {
+        arity_num_digits = 2;
+    } else {
+        arity_num_digits = 3;
+    }
+
     unsigned int module_name_len = atom_string_len(module);
     memcpy(buf, atom_string_data(module), module_name_len);
 
     buf[module_name_len] = ':';
 
     unsigned int function_name_len = atom_string_len(function);
-    if (UNLIKELY((arity > 9) || (module_name_len + function_name_len + 4 > buf_size))) {
+    if (UNLIKELY((arity > MAX_ARITY_VALUE) || ((module_name_len + function_name_len + 3 + arity_num_digits) > buf_size)))
+    {
         fprintf(stderr, "Insufficient room to write mfa.\n");
         abort();
     }
     memcpy(buf + module_name_len + 1, atom_string_data(function), function_name_len);
 
-    //TODO: handle functions with more than 9 parameters
-    buf[module_name_len + function_name_len + 1] = '/';
-    buf[module_name_len + function_name_len + 2] = '0' + arity;
-    buf[module_name_len + function_name_len + 3] = 0;
+    // Calculate the offset so we can get the pointer to the following unfilled position in the buffer
+    unsigned int offset = sizeof(char) * (module_name_len + function_name_len + 1);
```

Comment:
I think that if we use sizeof(char) then we need to use `size_t` for offset.
By the way, shouldn't we assume that char is always 1 byte in all modern C implementations?

Reply:
I don't really know. This is the first time I'm contributing C code outside of Uni, so I chose to stick to the safest path. If you'd rather assume sizeof(chat) = 1 byte always, I'd be happy to refactor this!